psi4/src/export_wavefunction.cc

```diff
@@ -0,0 +1,239 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * @END LICENSE
+ */
+
+#include "psi4/libmints/basisset.h"
+#include "psi4/libmints/sobasis.h"
+#include "psi4/libmints/molecule.h"
+#include "psi4/libmints/oeprop.h"
+#include "psi4/libmints/orbitalspace.h"
+
+#include "psi4/libfock/jk.h"
+#include "psi4/libfock/soscf.h"
+
+#include "psi4/detci/ciwave.h"
+#include "psi4/detci/civect.h"
+
+#include "psi4/libmints/mintshelper.h"
+#include "psi4/libmints/molecule.h"
+
+#include "psi4/libscf_solver/hf.h"
+#include "psi4/libscf_solver/rhf.h"
+#include "psi4/libscf_solver/uhf.h"
+#include "psi4/libscf_solver/rohf.h"
+#include "psi4/libscf_solver/cuhf.h"
+#include "psi4/libfunctional/superfunctional.h"
+#include "psi4/libfock/v.h"
+
+#include "psi4/libpsio/psio.h"
+#include "psi4/libpsio/psio.hpp"
+
+#include <string>
+
+using namespace psi;
+void export_wavefunction(py::module& m)
+{
+
+    typedef void (Wavefunction::*take_sharedwfn)(SharedWavefunction);
+    py::class_<Wavefunction, std::shared_ptr<Wavefunction>>(m, "Wavefunction", "docstring", py::dynamic_attr()).
+            def(py::init<std::shared_ptr<Molecule>, std::shared_ptr<BasisSet>, Options&>()).
+            def(py::init<std::shared_ptr<Molecule>, std::shared_ptr<BasisSet>>()).
+            def("reference_wavefunction", &Wavefunction::reference_wavefunction, "docstring").
+            def("set_reference_wavefunction", &Wavefunction::set_reference_wavefunction, "docstring").
+            def("shallow_copy", take_sharedwfn(&Wavefunction::shallow_copy), "docstring").
+            def("deep_copy", take_sharedwfn(&Wavefunction::deep_copy), "docstring").
+            def("same_a_b_orbs", &Wavefunction::same_a_b_orbs, "docstring").
+            def("same_a_b_dens", &Wavefunction::same_a_b_dens, "docstring").
+            def("nfrzc", &Wavefunction::nfrzc, "docstring").
+            def("nalpha", &Wavefunction::nalpha, "docstring").
+            def("nbeta", &Wavefunction::nbeta, "docstring").
+            def("nso", &Wavefunction::nso, "docstring").
+            def("nmo", &Wavefunction::nmo, "docstring").
+            def("nirrep", &Wavefunction::nirrep, "docstring").
+            def("Ca_subset", &Wavefunction::Ca_subset, py::return_value_policy::take_ownership, "docstring").
+            def("Cb_subset", &Wavefunction::Cb_subset, py::return_value_policy::take_ownership, "docstring").
```

Comment:
Do you want to also add a binding for `D_subset_helper` and/or `Da_subset`/`Db_subset`?

Those are probably useful to have and it would be consistent with `Ca_subset`/`Cb_subset` having bindings.